# setup

In [1]:
# dependencies
import pandas as pd

In [2]:
# support methods

In [3]:
# main
oemc = pd.read_parquet("../../data/OEMC_MP/export/output/oemc-prepped.parquet")

In [10]:
oemc['fishy_district'] = (oemc.numeric_district > 31) | (oemc.nonnumeric_district.notna())

# preview

In [11]:
oemc.sample().T

4142475
event_no                            1912003714
district                                   019
call_date                  2019-04-30 09:00:00
disp_date                  2019-04-30 09:13:00
on_date                                    NaT
clear_date                 2019-04-30 09:42:00
close_date                 2019-04-30 09:50:00
init_priority                               1A
init_type                        DEATH REMOVAL
fin_type                         DEATH REMOVAL
year_called                               2019
date_of_call                        2019-04-30
time_to_dispatch               0 days 00:13:00
ttd_group            Dispatch under 15 minutes
dispatch_reported                         True
closure_reported                          True
no_initial_type                          False
numeric_district                          19.0
nonnumeric_district                       None
init_on_view                             False
fin_on_view                              False
event_group                              other
event_type                               other
flagged                                  False
fishy_district                           False

# Frequency of fishy district values

In [12]:
oemc.loc[oemc.numeric_district > 30, 'numeric_district'].value_counts()

numeric_district
31.0     30891
377.0       56
177.0       35
61.0        33
477.0       25
41.0        11
70.0         3
Name: count, dtype: int64

In [13]:
oemc.loc[oemc.nonnumeric_district.notna(), 'nonnumeric_district'].value_counts()

nonnumeric_district
CW4     336551
CW5     193666
CW1     160144
CW3      68998
ORD       9614
CW2       8546
CW7       6708
CW6       3563
HD         290
FD         156
PT          53
ARS         17
OC          14
OEM         10
TST          7
PFDM         5
ARP          4
PPAG         3
SST          2
PHN          1
F02          1
F09          1
Name: count, dtype: int64

# final list + time covered

In [14]:
fishy_districts = oemc.loc[oemc.fishy_district, 'district'].unique()

In [16]:
for v in sorted(fishy_districts):
    print(f"- {v}")

- 061
- 070
- 177
- 377
- 4100
- 477
- ARP
- ARS
- CW1
- CW2
- CW3
- CW4
- CW5
- CW6
- CW7
- F02
- F09
- FD
- HD
- OC
- OEM
- ORD
- PFDM
- PHN
- PPAG
- PT
- SST
- TST


In [17]:
oemc.date_of_call.describe()

count       12159582
unique          1461
top       2021-07-04
freq           13647
Name: date_of_call, dtype: object

In [20]:
oemc.year_called.value_counts().sort_index().apply(lambda x: f"{x:,}")

year_called
2018    3,119,334
2019    3,293,586
2020    2,715,865
2021    3,030,797
Name: count, dtype: object

# Resulting FOIA request

I have previously requested and received emergency event data from OEMC covering 2018-2021 and am now requesting clarification about values I observe in the data so that I can make the appropriate data handling decisions.

The emergency events in the data each have an assigned "district" value which, because police are the most common agency dispatched, I assume refer to the police districts in the city's data portal (https://data.cityofchicago.org/Public-Safety/Boundaries-Police-Districts-current-/fthy-xz3r).

The district values in the data portal are all numeric and less than 35, but the following values appear in the OEMC data and do not match a district in the city portal or this pattern:
- 061
- 070
- 177
- 377
- 4100
- 477
- ARP
- ARS
- CW1
- CW2
- CW3
- CW4
- CW5
- CW6
- CW7
- F02
- F09
- FD
- HD
- OC
- OEM
- ORD
- PFDM
- PHN
- PPAG
- PT
- SST
- TST

Please provide clarification on these district values observed in the OEMC data. Let me know if I am comparing to the wrong city boundaries and these all appear in another city dataset, or if each of these match up to an existing police district and I should correct them in the data. If I should correct them, please advise what each value should be changed to.

Thank you for your time.